composer/callbacks/checkpoint_saver.py

```diff
@@ -45,11 +47,12 @@ def checkpoint_periodically(interval: Union[str, int, Time]) -> Callable[[State,
 
     if interval.unit == TimeUnit.EPOCH:
         save_event = Event.EPOCH_CHECKPOINT
-    elif interval.unit == TimeUnit.BATCH:
+    elif interval.unit in {TimeUnit.BATCH, TimeUnit.TOKEN, TimeUnit.SAMPLE}:
         save_event = Event.BATCH_CHECKPOINT
     else:
         raise NotImplementedError(
-            f'Unknown checkpointing interval: {interval.unit}. Must be TimeUnit.EPOCH or TimeUnit.BATCH.')
+            f'Unknown checkpointing interval: {interval.unit}. Must be TimeUnit.EPOCH, TimeUnit.BATCH, TimeUnit.TOKEN, or TimeUnit.SAMPLE.'
+        )
```

Comment:
Why not duration?

Reply:
well, it wasn't implemented before. We could use the same logic as for eval, but that seems like a separate PR?

Reply:
Hm... I feel like we should use the same logic. I'm fine doing in separate PR